services/user.service.js

```diff
@@ -59,45 +59,70 @@ class UserService extends BaseService {
       response = this.getErrorResponse(errorMessage)
     }
 
-    return response;
+    return response
   }
 
   async findById(id) {
     let response
 
     try {
-      const userRefSnapshot = await this.collection.doc(id).get();
+      const userRefSnapshot = await this.collection.doc(id).get()
 
       if (!userRefSnapshot.exists) {
-        return this.getSuccessResponse({}, 'No existing data');
+        return this.getSuccessResponse({}, 'No existing data')
       }
       const user = {
         ...userRefSnapshot.data(),
-        id: id
-      };
+        id: id,
+      }
 
-      const successMessage = 'Getting user information successfully';
-      response = this.getSuccessResponse(user, successMessage);
+      const successMessage = 'Getting user information successfully'
+      response = this.getSuccessResponse(user, successMessage)
     } catch (err) {
-      const errorMessage = 'Error getting user information';
+      const errorMessage = 'Error getting user information'
       /* eslint-disable no-console */
       // console.error(errorMessage, err);
       /* eslint-enable */
-      response = this.getErrorResponse(errorMessage);
+      response = this.getErrorResponse(errorMessage)
     }
 
     return response
   }
+  async checkUserAttendeeStatus(eventId, userId, eventsService) {
+    let response
+
+    try {
+      const eventData = await eventsService.findById(eventId)
+      if (!eventData) {
+        return this.getSuccessResponse(
+          { attendanceConfirmed: false },
+          'No existing data for this event'
+        )
+      }
+
+      const userData = eventData.data.attendees.find(attendee => attendee === userId)
```

Comment:
add checks for when data is null
```suggestion
      const userData = eventData.data?.attendees?.find(attendee => attendee === userId)
```

Reply:
Implementation pending to make it work